src/app/navigation/selectors.ts

```diff
@@ -38,4 +40,16 @@ export const match = createSelector(
   (state) => state.match
 );
 
+export const systemQueryParameters = createSelector(
+  query,
+  (navQuery) => pickBy(isDefined, {
+    'content-style': navQuery['content-style'] as string | undefined,
+  })
+);
+
+export const persistenQueryParameters = createSelector(
+  query,
+  (navQuery) => pickBy(isDefined, navQuery)
```

Comment:
i'm not sure i totally agree with all parameters being "persistent", i forget if i had listed a few before, but search definitely. the `modal` params i don't think really need to be here, because they aren't the kind of thing you have to reach across the app to persist on navigation without knowing what its for. if you're doing navigation in a modal and want to make sure it stays open, you'll be doing it in the modal